src/v/cloud_storage/remote.h

```diff
@@ -105,6 +106,11 @@ class auth_refresh_bg_op {
 /// things like reconnects, backpressure and backoff.
 class remote : public ss::peering_sharded_service<remote> {
 public:
+    /// Default tags applied to objects
+    static const std::vector<s3::object_tag> default_segment_tags;
```

Comment:
It's a bit inefficient that we are storing ss::sstring for the tag names, also formatting values into a sstring, but then eventually in request construction also using a stringstream to build the header string.

What if we define classes for each type of object, which just store values and then know how to generate the whole header string themselves?